.github/autobuild/mac.sh

```diff
@@ -30,20 +30,35 @@ prepare_signing() {
     [[ -n "${MACOS_CERTIFICATE:-}" ]] || return 1
     [[ -n "${MACOS_CERTIFICATE_ID:-}" ]] || return 1
     [[ -n "${MACOS_CERTIFICATE_PWD:-}" ]] || return 1
+    [[ -n "${MACAPP_CERTIFICATE:-}" ]] || return 1
+    [[ -n "${MACAPP_CERTIFICATE_ID:-}" ]] || return 1
+    [[ -n "${MACAPP_CERTIFICATE_PWD:-}" ]] || return 1
     [[ -n "${NOTARIZATION_PASSWORD:-}" ]] || return 1
     [[ -n "${KEYCHAIN_PASSWORD:-}" ]] || return 1
 
     echo "Signing was requested and all dependencies are satisfied"
 
-    # Put the cert to a file
-    echo "${MACOS_CERTIFICATE}" | base64 --decode > certificate.p12
+    ## Put the certs to a file
+    # MACOS_CERTIFICATE - Mac Installer Distribution
+    # MACAPP_CERTIFICATE - Mac App Distribution
+    echo "${MACOS_CERTIFICATE}" | base64 --decode > macinst_certificate.p12
+    echo "${MACAPP_CERTIFICATE}" | base64 --decode > macapp_certificate.p12
 
     # Set up a keychain for the build:
     security create-keychain -p "${KEYCHAIN_PASSWORD}" build.keychain
     security default-keychain -s build.keychain
     security unlock-keychain -p "${KEYCHAIN_PASSWORD}" build.keychain
-    security import certificate.p12 -k build.keychain -P "${MACOS_CERTIFICATE_PWD}" -T /usr/bin/codesign
-    security set-key-partition-list -S apple-tool:,apple:,codesign: -s -k "${KEYCHAIN_PASSWORD}" build.keychain
+    security import macinst_certificate.p12 -k build.keychain -P "${MACOS_CERTIFICATE_PWD}" -A -T /usr/bin/productbuild
+    security import macapp_certificate.p12 -k build.keychain -P "${MACAPP_CERTIFICATE_PWD}" -A -T /usr/bin/codesign 
+    # security set-key-partition-list -S apple-tool:,apple:,codesign: -s -k "${KEYCHAIN_PASSWORD}" build.keychain
+    security set-key-partition-list -S apple-tool:,apple: -s -k "${KEYCHAIN_PASSWORD}" build.keychain
+
+    # set lock timeout on keychain to 6 hours
+    security set-keychain-settings -lut 21600
```

Comment:
Why 6 h? Probably due to automatic lock and a resulting failure. Is there a cleaner way? Why is this needed now and wasn't previously?

Reply:
Good question. One of the hardest parts in getting this to work was working out how and why either `codesign` or `productbuild` would hang, likely due to prompting interactively for authentication.
Apple does a terrible job at documenting the necessary arguments for:
`  security set-key-partition-list -S ...`
and the effectiveness of the -A and -T options on `security import` had to be worked out through trial and error.
It's possible that this extended timeout is redundant, but it is apparently a reasonable setting for similar project builds, especially as these are being run on ephemeral hosted Github Action runners.

Reply:
I think this part (line 62 & 63) is redundant by now as we have removed the re-locking timeout altogether: https://github.com/jamulussoftware/jamulus/pull/2927/files

Reply:
I'd like to test if the CodeQl issue is fixed with this instead.

Reply:
I guess it's worth a test, but I wouldn't expect too much based on the docs and comments elsewhere:
https://ss64.com/osx/security-keychain-settings.html
https://developer.apple.com/forums/thread/690665

Reply:
removed the redundant timeout setting